dotCMS/src/main/java/com/dotcms/rest/SearchForm.java

```diff
@@ -0,0 +1,98 @@
+package com.dotcms.rest;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+/**
+ * Search Form to make a ES query
+ * @author jsanca
+ */
+@JsonDeserialize(builder = SearchForm.Builder.class)
+public class SearchForm {
+
+    private final String query;
+    private final String sort;
+    private final int limit;
+    private final int offset;
+    private final String userId;
+
+    private SearchForm (final Builder builder) {
+
+        this.query  = builder.query;
+        this.sort   = builder.sort;
+        this.limit  = builder.limit;
+        this.offset = builder.offset;
+        this.userId = builder.userId;
+    }
+
+    public String getQuery() {
+        return query;
+    }
+
+    public String getSort() {
+        return sort;
+    }
+
+    public int getLimit() {
+        return limit;
+    }
+
+    public int getOffset() {
+        return offset;
+    }
+
+    public String getUserId() {
+        return userId;
+    }
+
+    @Override
+    public String toString() {
+        return "SearchForm{" +
+                "query='" + query + '\'' +
+                ", sort='" + sort + '\'' +
+                ", limit=" + limit +
+                ", offset=" + offset +
+                ", userId='" + userId + '\'' +
+                '}';
+    }
+
+    public static final class Builder {
+
+        private  @JsonProperty String query  = "";
+        private  @JsonProperty String sort   = "";
```

Comment:
Codacy found an issue: [Field sort has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6233451)